### DISCUSSION TITLE
Iron only top and bottom layer and/or choose ironing layers

### DISCUSSION THREAD

**Discussion by freakybuckey** [OrcaSlicer/OrcaSlicer#1389] [Ideas]
Sometimes I'd like to iron both the first layer (in order to remove any artifacts where adjacent lines meet) and the top layer, but the only way to do this at present is to select ironing for "All solid layers".  This ends up causing the shell layers above the bottom layer to also be ironed, adding unnecessary print time.  An option to iron only the top and bottom layers would be helpful.  

Alternatively, the ability to choose specific layers to iron would work.

**Comment by pemerick07** [comment]
The ability to iron specific layers would be an excellent feature. I iron the bottom of o-ring grooves and the top of flange faces, but don't need any other surface iron. You should put this in "Issues" as a feature request if you haven't already.

**Comment by freakybuckey** [comment]
Thanks, I'm adding it there now.  I'm new to github so I apologize if I've posted this in the wrong place.

**Comment by tripp396** [comment]
Not to resurrect an old thread but I was searching for this as a feature request and then realized I could accomplish this via height range modifiers on the object for the specific layer, and just adjust ironing on that modifier.

**Reply by ajax-pyvovarov-y** [reply]
Could you elaborate on it?

**Reply by matulek** [reply]
Actually I can because I just wanted to do the exact same thing. You can go to "objects" menu here:

<img width="630" height="613" alt="image" src="https://github.com/user-attachments/assets/87a843f0-e7af-4a78-a7be-f68808004642" />

and you can add "height range modifier" for layers at specific heights. In there you can change any setting you wish, including enabling ironing "for all solid layers" (it'll work for the specific range).

**Comment by ndbills** [comment]
This is great, and I really like this idea. I've used height range modifiers to set ironing at specific layers, but even beyond choosing a layer to iron, if specific surfaces could be "painted" for ironing (similar to how you can paint to add/remove supports) would be SO awesome. There are cases where I want to iron part of a layer, but not the entire layer, and could save significant time in printing if ironing could be more selective. Does that make sense?

**Reply by tripp396** [reply]
Ohhh I really like the idea of “painting” top surfaces for ironing!

**Reply by skl111** [reply]
I came upon this thread looking for this, as ironing works great except on some small features I have on a model where it destroys them. For now I will look into height range modifiers.
